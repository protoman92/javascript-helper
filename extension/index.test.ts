import ".";

describe("Array extensions", () => {
  it("First and last should work correctly", () => {
    // Setup && When && Then
    expect([1, 2].first()).toEqual(1);
    expect([1, 2].last()).toEqual(2);
    expect([].last()).toEqual(undefined);
  });

  it("Compact map should work correctly", () => {
    // Setup
    const arr = [1, undefined, 2, null, 3];

    // When && Then
    expect(arr.compactMap()).toEqual([1, 2, 3]);
    expect(arr.compactMap(() => undefined)).toEqual([]);
    expect(arr.compactMap()).not.toEqual(arr);
  });

  it("Merge elements should work", () => {
    // Setup
    const arr = [{ a: 1 }, { b: 2 }, { c: 3 }];

    // When
    expect(arr.mergeElements()).toEqual({ a: 1, b: 2, c: 3 });
  });
});
