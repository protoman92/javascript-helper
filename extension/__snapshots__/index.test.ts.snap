// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array extensions toMap should work 1`] = `
Map {
  1 => Object {
    "a": 1,
  },
  2 => Object {
    "a": 2,
  },
  3 => Object {
    "a": 3,
  },
}
`;

exports[`Array extensions toMap should work 2`] = `
Map {
  1 => Object {
    "a": 1,
  },
  2 => Object {
    "a": 2,
  },
  3 => Object {
    "a": 3,
  },
}
`;

exports[`Array extensions toMap should work 3`] = `
Map {
  "1" => Object {
    "a": 1,
  },
  "2" => Object {
    "a": 2,
  },
  "3" => Object {
    "a": 3,
  },
}
`;
